src/Spectre.Angular2Client/src/app/heatmap/heatmap.component.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * heatmap.component.ts
+ * Component of single heatmap.
+ *
+ Copyright 2017 Grzegorz Mrukwa, Sebastian Pustelnik, Daniel Babiak
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import {Component, Input, OnInit} from '@angular/core';
+import {HeatmapService} from './heatmap.service';
+import {Heatmap} from './heatmap';
+
+
+
+@Component({
+  selector: 'app-heatmap',
+  templateUrl: './heatmap.component.html',
+  styleUrls: ['./heatmap.component.css']
+})
+export class HeatmapComponent implements OnInit {
+
+  @Input() public id: number;
```

Comment:
Besides of removing `heatmapService` from here, few more refinements are necessary. To be precise:

* If we do not hide layout and other options from the user, they should be typed in such a way, that no value set breaks the component, even if we limit the functionality much (for now we do not need much)
* Same situation is with the `HeatmapData`. There is an option to make it a `Heatmap` (this interface we already have), since whole translation is inside. No one will be able to break this component providing bad data.

To avoid too much work in this task (since it is not crucial to do all this stuff with all layout parameters, etc.), I would temporarily restrict modification of anything but data from the outside. Just to make it unbreakable.